src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -847,6 +847,26 @@ class TypedStructLiteralSegment(BaseSegment):
     )
 
 
+class EmptyStructLiteralBracketsSegment(BaseSegment):
+    """An empty struct literal segment - `()`.
+
+    NOTE: This is only to set the right type so spacing rules are applied correctly.
```

Comment:
Given this, it might be a good idea to also add a test to [test/fixtures/rules/std_rule_cases/LT01-excessive.yml](https://github.com/sqlfluff/sqlfluff/blob/main/test/fixtures/rules/std_rule_cases/LT01-excessive.yml) ?

Reply:
I added a new test on [CP05.yml](https://github.com/sqlfluff/sqlfluff/pull/4879/files#diff-3c4316e5a2b25d65eaa58fe8a0238fc6ce538890fbc00fedcc8f0533614179d1). I couldn't figure out what test I should add on LT01. Let me know if I should do it otherwise